src/instructlab/train/linux_train.py

```diff
@@ -155,9 +168,14 @@ def linux_train(
         model_name, torchscript=True, trust_remote_code=True
     )
 
+    model_torch_dtype = "auto"
```

Comment:
So IIUC, the "auto" is not capable of tuning the behavior based on CPU or GPU? Do you know what does it do then?

Reply:
It looks like it selects bfloat16. This is a terrible choice for my laptop and I'd guess most non-server CPUs as well as it triggers single threaded execution.